### PR TITLE
Unify the metrics for queuesize similar to what we have for the spi q…

### PR DESCRIPTION
**Pull request by baldersheim**
…ueues.

@geirst and @vekterli PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@yngveaasheim I have added better metrics for the executor queues in the backend.
They are similar to what you have for the  spi operations Q.

**Review by vekterli — APPROVED** [review]
:+1:

File: searchcore/src/vespa/searchcore/proton/metrics/executor_metrics.cpp [hunk 1]
```diff
@@ -7,16 +7,19 @@ namespace proton {
 void
 ExecutorMetrics::update(const vespalib::ThreadStackExecutorBase::Stats &stats)
 {
-    maxPending.set(stats.maxPendingTasks);
+    maxPending.set(stats.queueSize.max());
     accepted.inc(stats.acceptedTasks);
     rejected.inc(stats.rejectedTasks);
+    const vespalib::ThreadStackExecutorBase::Stats::QueueSizeT & qSize = stats.queueSize;
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/metrics/executor_metrics.cpp]: Consider using auto.

File: searchcore/src/vespa/searchcore/proton/metrics/executor_metrics.cpp [hunk 2]
```diff
@@ -7,16 +7,19 @@ namespace proton {
...
+    queueSize.addValueBatch(qSize.average(), qSize.count(), qSize.min(), qSize.max());
 }
 
 ExecutorMetrics::ExecutorMetrics(const std::string &name, metrics::MetricSet *parent)
     : metrics::MetricSet(name, {}, "Instance specific thread executor metrics", parent),
       maxPending("maxpending", {}, "Maximum number of pending (active + queued) tasks", this),
       accepted("accepted", {}, "Number of accepted tasks", this),
-      rejected("rejected", {}, "Number of rejected tasks", this)
+      rejected("rejected", {}, "Number of rejected tasks", this),
+      queueSize("queuesize", {}, "size of task queue", this)
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/metrics/executor_metrics.cpp]: "size" -> "Size"

**Review by geirst — COMMENTED** [review]
👍

File: searchcore/src/vespa/searchcore/proton/metrics/executor_metrics.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/metrics/executor_metrics.cpp]: Fixed in https://github.com/vespa-engine/vespa/pull/12711/commits/a4f448633278360f7cb8ef2135f68291cb021a9f

File: searchcore/src/vespa/searchcore/proton/metrics/executor_metrics.cpp (hunk 2 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/metrics/executor_metrics.cpp]: Fixed in https://github.com/vespa-engine/vespa/pull/12711/commits/a4f448633278360f7cb8ef2135f68291cb021a9f

**Pull request merged**